packages/component-base/src/polylit-mixin.js

```diff
@@ -214,6 +220,11 @@ const PolylitMixinImplementation = (superclass) => {
         parentHost.$ ||= {};
         parentHost.$[this.id] = this;
       }
+
+      const { polylitConfig } = this.constructor;
+      if (!polylitConfig.asyncFirstRender) {
+        this.performUpdate();
```

Comment:
I wonder if we should check for the actual first render here? With the current implementation, this is called on every `connectedCallback()` which means it will be also executed on moving element to the new parent (can happen e.g. during overlay teleportation).

Reply:
Good question. Does Polymer render only once?

Reply:
Right, we probably could add a test for this to ensure that it would render synchronously in that case as well.

Reply:
Based on my testing, both Lit and Polymer still re-render after the component is disconnected, so it indeed makes sense to limit `performUpdate` to the first render only.

Reply:
Done.